### PR TITLE
Feature/refactor user

### PR DESCRIPTION
**Pull request by Friederi**
Check check!

### CODE REVIEW

File: tech-project-back-end/Services/IService/IUserService.cs [hunk 1]
```diff
@@ -1,7 +1,20 @@
...
     {
         public Task<int> GetTotalUser();
+        public Task<List<UserDTO>> GetAllUser();
+        public Task<UserDTO> GetUserById(string userid);
+        public Task<(bool Success, string? Message, UserDTO? User, string? Token)> Login(UserLoginDTO userLogin);
+        public Task<(bool Success, string? Message, UserDTO? User)> UpdateUser(UserUpdateDTO updatedUser);
+        public Task<(bool Success, string? Message)> ForgetPassword(string email);
+        public Task<(bool Success, string? Message, UserDTO? User, string? Token)> Register(UserDTO userRegisterDto);
+        public Task<string> CreateToken(User user);
+        public string GetEmailTemplate(string templateName);
```
Comment (hSangg) [inline tech-project-back-end/Services/IService/IUserService.cs]: Is this method using in another component?. If not. Please dont defined in IService

**Review by hSangg — CHANGES_REQUESTED** [review]
Please check

File: tech-project-back-end/DTO/Users/UserUpdateDTO.cs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+﻿namespace tech_project_back_end.DTO.Users
+{
+    public class UserUpdateDTO
+    {
+        public string UserId { get; set; }
+        public string Phone { get; set; }
+        public string Email { get; set; }
+        public string Name { get; set; }
+
```
Comment (hSangg) [inline tech-project-back-end/DTO/Users/UserUpdateDTO.cs:10]: carefully check unused space

File: tech-project-back-end/Repository/IRepository/IUserRepository.cs
Comment (Cu-Hoang) [inline tech-project-back-end/Repository/IRepository/IUserRepository.cs:1]: chưa tách dòng

File: tech-project-back-end/Services/IService/IUserService.cs
Comment (Cu-Hoang) [inline tech-project-back-end/Services/IService/IUserService.cs:1]: chưa tách dòng

File: tech-project-back-end/Controllers/UserController.cs [hunk 3]
```diff
@@ -19,225 +18,154 @@ public class UserController : ControllerBase
         private readonly AppDbContext _appDBContext;
```
Comment (Cu-Hoang) [inline tech-project-back-end/Controllers/UserController.cs]: thuộc tính ko sử dụng có thể xóa

File: tech-project-back-end/Services/IService/IUserService.cs (hunk 1 above)
Comment (Friederi) [reply tech-project-back-end/Services/IService/IUserService.cs]: For the two GetEmailTemplate and ReplacePlaceholders methods, moving them EmailService while CreateToken remains the same. Or do you propose creating a Helper class in Utilities to handle miscellaneous tasks like CreateToken?

Comment (hSangg) [reply tech-project-back-end/Services/IService/IUserService.cs]: my main idea is if this method is only using inside its service. you shouldnt define it int Iservice

File: tech-project-back-end/Services/IService/IUserService.cs
Comment (hSangg) [reply tech-project-back-end/Services/IService/IUserService.cs:1]: Where does the line that you imply?

**Review by hSangg — APPROVED** [review]
Good em

**Review by KietGia22 — APPROVED** [review]
lỗi thì Công sửa

**Pull request merged**